platforms/documentation/docs/build.gradle.kts

```diff
@@ -85,24 +85,47 @@ dependencies {
 
 asciidoctorj {
     setVersion("2.5.13")
-    modules {
-        pdf.use()
-        pdf.setVersion("2.3.10")
-    }
+    modules.pdf.setVersion("2.3.10")
     // TODO: gif are not supported in pdfs, see also https://github.com/gradle/gradle/issues/24193
     // TODO: tables are not handled properly in pdfs
     fatalWarnings.add(
         Regex("^(?!GIF image format not supported|dropping cells from incomplete row detected end of table|.*Asciidoctor PDF does not support table cell content that exceeds the height of a single page).*").toPattern()
     )
 }
 
-tasks.withType<AsciidoctorTask>().configureEach {
-    if (name == "userguideSinglePagePdf") {
-        asciidoctorj.docExtensions(
+tasks.named<AsciidoctorTask>("userguideSinglePagePdf").configure {
+    val task = this
+    task.extensions.getByType<org.asciidoctor.gradle.jvm.AsciidoctorJExtension>().apply {
+        docExtensions(
             project.dependencies.create(project(":docs-asciidoctor-extensions-base"))
         )
-    } else {
-        asciidoctorj.docExtensions(
+    }
+}
+
+tasks.named<AsciidoctorTask>("userguideSinglePageHtml").configure  {
+    val task = this
+    task.extensions.getByType<org.asciidoctor.gradle.jvm.AsciidoctorJExtension>().apply {
```

Comment:
For something more standard like `extensions` we probably don't need to qualify it with a `task.`